### ISSUE TITLE
Wrong zip format generated for repacking 'kotlin-compiler-embeddable-1.3.72.jar'

### ISSUE THREAD

**Issue by devkanro** [gradle/gradle#13526]
### Expected Behavior
Nothing went to wrong.

### Current Behavior
We got a build error after upgrade to 6.5.

### Context
There is the error message
```log
* What went wrong:
A problem occurred configuring root project 'sisyphus'.
> Could not open cache directory 7hif95utaztsqa06tlx129py9 (C:\Users\higan\.gradle\caches\6.5\gradle-kotlin-dsl\7hif95utaztsqa06tlx129py9).
   > Could not open cache directory ccpwzn58qqes1cuocloljg8d (C:\Users\higan\.gradle\caches\6.5\gradle-kotlin-dsl-accessors\ccpwzn58qqes1cuocloljg8d).
      > malformed input off : 456328, length : 1
```

After some breakpoints, I think there is some wrong in Gradle with 'kotlin-compiler-embeddable-1.3.72.jar'.
The exception occurs when getting the name of jar entry, some non-UTF-8 characters are included in 'kotlin-compiler-embeddable-1.3.72.jar'.

I think the 'λ.class' is the suspect.

the bad `$HOMT\.gradle\caches\jars-8\3b30f84b872ed1dd3fb7e12bb39d11c0\kotlin-compiler-embeddable-1.3.72.jar`
![Snipaste_2020-06-19_16-29-18](https://user-images.githubusercontent.com/9367842/85114107-fcfd3c80-b24a-11ea-9d44-de095b181b24.png)

the good `kotlin-compiler-embeddable-1.3.72.jar`
![Snipaste_2020-06-19_16-38-15](https://user-images.githubusercontent.com/9367842/85114242-4e0d3080-b24b-11ea-9c0f-f1ed3fb59747.png)

It seems Gradle will unpack the `kotlin-compiler-embeddable-1.3.72.jar`, and repack it in 6.5
Some string encoding issue occurred when repacking.

### Steps to Reproduce 
01. Clone the master branch of [ButterCam/sisyphus](https://github.com/ButterCam/sisyphus) project
02. **REQUIRED** Configure JDK11
03. Upgrade the Gradle to 6.5
04. Refresh project

### Your Environment
Windows 10 1909
Gradle 6.5

**Comment by devkanro** [comment]
It seems this PR #13359 cause the exception?
@wolfs

**Comment by devkanro** [comment]
@wolfs 
After some research, I found the `~/.gradle/caches/jars-8` folder is the dependencies cache of the plugin, maybe?
All jars will be repacked, and the 'kotlin-compiler-embeddable-1.3.72.jar' has some Unicode file name in it.

Before Gradle 6.5, the 'kotlin-compiler-embeddable-1.3.72.jar' will not be generated in `~/.gradle/caches/jars-8` folder.

**Comment by wolfs** [comment]
@devkanro I suppose you were able to fix your problem? https://github.com/gradle/gradle/pull/13359 shouldn't have anything to do with it since this is only for the `gradle/gradle` project and won't have any effect on you using Gradle 6.5.

**Comment by devkanro** [comment]
@wolfs 
I found that I can reproduce in some PCs, but others not, I don't know how it happened, so I close it.
I got a workaround by replacing the jar in the 'jar-8' folder.

**Comment by znjameswu** [comment]
I'm experiencing the exact same problem. The same `gradle-kotlin-dsl` and `gradle-kotlin-dsl-accessor`.

**Comment by wolfs** [comment]
@znjameswu You might be hitting something similar to https://github.com/gradle/gradle/issues/14155, which has been fixed in 6.7. Can you see if 6.7 fixes your problem as well?

**Comment by znjameswu** [comment]
> @znjameswu You might be hitting something similar to #14155, which has been fixed in 6.7. Can you see if 6.7 fixes your problem as well?

Yes I upgrade to 6.7.1, removed .gradle/caches and solved this problem.
